frontend/lib/resolution/resolution-error-classes-list.cpp

```diff
@@ -615,6 +616,53 @@ void ErrorInvalidClassCast::write(ErrorWriterBase& wr) const {
   }
 }
 
+void ErrorInvalidDomainCall::write(ErrorWriterBase& wr) const {
+  auto fnCall = std::get<const uast::FnCall*>(info_);
+  auto actualTypes = std::get<std::vector<types::QualifiedType>>(info_);
+
+  wr.heading(kind_, type_, fnCall, "invalid use of the 'domain' keyword.");
+  wr.codeForLocation(fnCall);
+  wr.message(
+      "The 'domain' keyword should be used with a valid domain type "
+      "expression.");
+
+  if (fnCall->numActuals() == 0) {
+    wr.message("However, 'domain' here did not have any actuals.");
+  } else if (fnCall->numActuals() == 1) {
+    // Could be rectangular or associative. Error if we have an actual type
+    // that's wrong for both.
+    auto qt = actualTypes[0];
+    if (!qt.isType() && !(qt.type() && qt.type()->isIntType())) {
+      wr.message(
+          "However, the first actual was ", decayToValue(qt),
+          " rather than an int (for rectangular) or a type (for associative).");
```

Comment:
Got it. Does "type" here also count as Chapel syntax? I could see it either way since `type` is an intent used in Chapel syntax, but also just a concept

Reply:
Got it. Does "type" here also count as Chapel syntax? I could see it either way since `type` is an intent used in Chapel syntax, but also just a concept

Reply:
I think in this case, it's used as the concept, so I would not quote it -- which seems to have been your intuition as well.